api/users/views.py

```diff
@@ -55,25 +56,37 @@ def get_queryset(self):
         return User.find(query)
 
 
-class UserDetail(generics.RetrieveAPIView, UserMixin):
+class UserDetail(generics.RetrieveUpdateAPIView, UserMixin):
     """Details about a specific user.
     """
+    permission_classes = (
+        ReadOnlyOrCurrentUser,
+    )
     serializer_class = UserSerializer
 
     # overrides RetrieveAPIView
     def get_object(self):
         return self.get_user()
 
+    # overrides RetrieveUpdateAPIView
+    def get_serializer_context(self):
```

Comment:
From you, all right? I learned it by watching you!

Okay, we'll just duplicate that code everywhere. That's cool.